### PR TITLE
new kusto upload to AzNhcRunEvents

### PR DESCRIPTION
**Pull request by shivanispatel**
- parse for test results from CPU or GPU (include error checking)
- results are outputted in JSON format to jsonResult column

### CODE REVIEW

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 1]
```diff
@@ -60,13 +63,177 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
...
+            'workflowType': "main",
+            'time': ts,
+            'pass': False, # keep as default false
+            'error': '',
+            'logOutput': full_results, # the entire file
+            'jsonResult': jsonResult,
+            'uuid': full_uuid
+        }
+
+        if "Node Health Check completed successfully " in full_results:
```
Comment (yakovdyadkin) [inline distributed_nhc/export_nhc_result_to_kusto.py]: Do you need the whitespace after "successfully"?

Comment (shivanispatel) [reply distributed_nhc/export_nhc_result_to_kusto.py]: Won't cause issues as there's some things that show up afterwards, but don't need it; removed!

**Review by yakovdyadkin — APPROVED** [review]
One non-critical comment + offline discussion

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 2]
```diff
@@ -60,13 +63,177 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
...
+            if key.startswith("ib_write_lb_mlx5_ib"):
+                result["IB_WRITE_GDR"][key] = str(value.strip())
+            elif key.startswith("H2D"):
+                result["GPU_BW_HTD"][key] = str(value.strip())
+            elif key.startswith("D2H"):
+                result["GPU_BW_DTH"][key] = str(value.strip())
+            elif key.startswith("P2P"):
+                result["GPU_BW_P2P"][key] = str(value.strip())
+            elif key.startswith("nccl_all_red"):
+                result["NCCL_ALL_REDUCE"] = str(value.strip())
```
Comment (rafsalas19) [inline distributed_nhc/export_nhc_result_to_kusto.py]: forgot to add [key]

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 3]
```diff
@@ -60,13 +63,177 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
...
+            elif key.startswith("nccl_all_red_lb"):
+                result["NCCL_ALL_REDUCE_LOOP_BACK"] = str(value.strip())
```
Comment (rafsalas19) [inline distributed_nhc/export_nhc_result_to_kusto.py:119]: forgot to add [key]

Comment (shivanispatel) [reply distributed_nhc/export_nhc_result_to_kusto.py:119]: Not adding key, but have fixed the issue (it was reading "nccl_all_red" first, so it was actually grabbing the "nccl_all_red_lb" value instead, causing the issue). Flipping the order fixed the issue.

**Comment by shivanispatel** [comment]
If the parsed values are none, the upload accounts for this. Handles empty lines, white space only lines, and None / null values

**Review by yakovdyadkin — APPROVED** [review]

**Pull request merged**